File: app/src/main/java/ValidateUser/InputValidationRegex.java
package ValidateUser;

import java.util.regex.Pattern;

public class RegexValidation {
    public static boolean hasNumeric(String input){
        return Pattern.compile("[0-9]").matcher(input).find();
    }
    public static boolean hasSpecialCharacters(String input){
        return Pattern.compile("[!#$%&*()_+=|<>?{}\\[\\]~]").matcher(input).find();
    }
}

File: app/src/main/java/ValidateUser/InputValidationRegex.kt
package ValidateUser

import java.util.regex.Pattern

object InputValidationRegex {
    fun hasNumeric(input: String): Boolean {
        return Pattern.compile("[0-9]").matcher(input).find()
    }

    fun hasSpecialCharacters(input: String): Boolean {
        return Pattern.compile("[!#$%&*()_+=|<>?{}\\[\\]~]").matcher(input).find()
    }

}